// @nibiruchain/protojs/dist/perp/v1/genesis.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenesisState = exports.protobufPackage = void 0;
/* eslint-disable */
const state_1 = require("./state");
const long_1 = __importDefault(require("long"));
const _m0 = __importStar(require("protobufjs/minimal"));
exports.protobufPackage = "nibiru.perp.v1";
function createBaseGenesisState() {
    return { params: undefined, pairMetadata: [], positions: [], prepaidBadDebts: [] };
}
exports.GenesisState = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.params !== undefined) {
            state_1.Params.encode(message.params, writer.uint32(10).fork()).ldelim();
        }
        for (const v of message.pairMetadata) {
            state_1.PairMetadata.encode(v, writer.uint32(18).fork()).ldelim();
        }
        for (const v of message.positions) {
            state_1.Position.encode(v, writer.uint32(26).fork()).ldelim();
        }
        for (const v of message.prepaidBadDebts) {
            state_1.PrepaidBadDebt.encode(v, writer.uint32(34).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGenesisState();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.params = state_1.Params.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.pairMetadata.push(state_1.PairMetadata.decode(reader, reader.uint32()));
                    break;
                case 3:
                    message.positions.push(state_1.Position.decode(reader, reader.uint32()));
                    break;
                case 4:
                    message.prepaidBadDebts.push(state_1.PrepaidBadDebt.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            params: isSet(object.params) ? state_1.Params.fromJSON(object.params) : undefined,
            pairMetadata: Array.isArray(object === null || object === void 0 ? void 0 : object.pairMetadata)
                ? object.pairMetadata.map((e) => state_1.PairMetadata.fromJSON(e))
                : [],
            positions: Array.isArray(object === null || object === void 0 ? void 0 : object.positions)
                ? object.positions.map((e) => state_1.Position.fromJSON(e))
                : [],
            prepaidBadDebts: Array.isArray(object === null || object === void 0 ? void 0 : object.prepaidBadDebts)
                ? object.prepaidBadDebts.map((e) => state_1.PrepaidBadDebt.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        message.params !== undefined &&
            (obj.params = message.params ? state_1.Params.toJSON(message.params) : undefined);
        if (message.pairMetadata) {
            obj.pairMetadata = message.pairMetadata.map((e) => e ? state_1.PairMetadata.toJSON(e) : undefined);
        }
        else {
            obj.pairMetadata = [];
        }
        if (message.positions) {
            obj.positions = message.positions.map((e) => (e ? state_1.Position.toJSON(e) : undefined));
        }
        else {
            obj.positions = [];
        }
        if (message.prepaidBadDebts) {
            obj.prepaidBadDebts = message.prepaidBadDebts.map((e) => e ? state_1.PrepaidBadDebt.toJSON(e) : undefined);
        }
        else {
            obj.prepaidBadDebts = [];
        }
        return obj;
    },
    fromPartial(object) {
        var _a, _b, _c;
        const message = createBaseGenesisState();
        message.params =
            object.params !== undefined && object.params !== null
                ? state_1.Params.fromPartial(object.params)
                : undefined;
        message.pairMetadata =
            ((_a = object.pairMetadata) === null || _a === void 0 ? void 0 : _a.map((e) => state_1.PairMetadata.fromPartial(e))) || [];
        message.positions = ((_b = object.positions) === null || _b === void 0 ? void 0 : _b.map((e) => state_1.Position.fromPartial(e))) || [];
        message.prepaidBadDebts =
            ((_c = object.prepaidBadDebts) === null || _c === void 0 ? void 0 : _c.map((e) => state_1.PrepaidBadDebt.fromPartial(e))) || [];
        return message;
    },
};
if (_m0.util.Long !== long_1.default) {
    _m0.util.Long = long_1.default;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
