// @nibiruchain/protojs/dist/spot/v1/params.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Params = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const coin_1 = require("../../cosmos/base/v1beta1/coin");
const _m0 = __importStar(require("protobufjs/minimal"));
exports.protobufPackage = "nibiru.spot.v1";
function createBaseParams() {
    return { startingPoolNumber: long_1.default.UZERO, poolCreationFee: [], whitelistedAsset: [] };
}
exports.Params = {
    encode(message, writer = _m0.Writer.create()) {
        if (!message.startingPoolNumber.isZero()) {
            writer.uint32(8).uint64(message.startingPoolNumber);
        }
        for (const v of message.poolCreationFee) {
            coin_1.Coin.encode(v, writer.uint32(18).fork()).ldelim();
        }
        for (const v of message.whitelistedAsset) {
            writer.uint32(26).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.startingPoolNumber = reader.uint64();
                    break;
                case 2:
                    message.poolCreationFee.push(coin_1.Coin.decode(reader, reader.uint32()));
                    break;
                case 3:
                    message.whitelistedAsset.push(reader.string());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            startingPoolNumber: isSet(object.startingPoolNumber)
                ? long_1.default.fromValue(object.startingPoolNumber)
                : long_1.default.UZERO,
            poolCreationFee: Array.isArray(object === null || object === void 0 ? void 0 : object.poolCreationFee)
                ? object.poolCreationFee.map((e) => coin_1.Coin.fromJSON(e))
                : [],
            whitelistedAsset: Array.isArray(object === null || object === void 0 ? void 0 : object.whitelistedAsset)
                ? object.whitelistedAsset.map((e) => String(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        message.startingPoolNumber !== undefined &&
            (obj.startingPoolNumber = (message.startingPoolNumber || long_1.default.UZERO).toString());
        if (message.poolCreationFee) {
            obj.poolCreationFee = message.poolCreationFee.map((e) => e ? coin_1.Coin.toJSON(e) : undefined);
        }
        else {
            obj.poolCreationFee = [];
        }
        if (message.whitelistedAsset) {
            obj.whitelistedAsset = message.whitelistedAsset.map((e) => e);
        }
        else {
            obj.whitelistedAsset = [];
        }
        return obj;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseParams();
        message.startingPoolNumber =
            object.startingPoolNumber !== undefined && object.startingPoolNumber !== null
                ? long_1.default.fromValue(object.startingPoolNumber)
                : long_1.default.UZERO;
        message.poolCreationFee =
            ((_a = object.poolCreationFee) === null || _a === void 0 ? void 0 : _a.map((e) => coin_1.Coin.fromPartial(e))) || [];
        message.whitelistedAsset = ((_b = object.whitelistedAsset) === null || _b === void 0 ? void 0 : _b.map((e) => e)) || [];
        return message;
    },
};
if (_m0.util.Long !== long_1.default) {
    _m0.util.Long = long_1.default;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
