// @nibiruchain/protojs/dist/stablecoin/params.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Params = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const _m0 = __importStar(require("protobufjs/minimal"));
exports.protobufPackage = "nibiru.stablecoin.v1";
function createBaseParams() {
    return {
        collRatio: long_1.default.ZERO,
        feeRatio: long_1.default.ZERO,
        efFeeRatio: long_1.default.ZERO,
        bonusRateRecoll: long_1.default.ZERO,
        distrEpochIdentifier: "",
        adjustmentStep: long_1.default.ZERO,
        priceLowerBound: long_1.default.ZERO,
        priceUpperBound: long_1.default.ZERO,
        isCollateralRatioValid: false,
    };
}
exports.Params = {
    encode(message, writer = _m0.Writer.create()) {
        if (!message.collRatio.isZero()) {
            writer.uint32(8).int64(message.collRatio);
        }
        if (!message.feeRatio.isZero()) {
            writer.uint32(16).int64(message.feeRatio);
        }
        if (!message.efFeeRatio.isZero()) {
            writer.uint32(24).int64(message.efFeeRatio);
        }
        if (!message.bonusRateRecoll.isZero()) {
            writer.uint32(32).int64(message.bonusRateRecoll);
        }
        if (message.distrEpochIdentifier !== "") {
            writer.uint32(42).string(message.distrEpochIdentifier);
        }
        if (!message.adjustmentStep.isZero()) {
            writer.uint32(48).int64(message.adjustmentStep);
        }
        if (!message.priceLowerBound.isZero()) {
            writer.uint32(56).int64(message.priceLowerBound);
        }
        if (!message.priceUpperBound.isZero()) {
            writer.uint32(64).int64(message.priceUpperBound);
        }
        if (message.isCollateralRatioValid === true) {
            writer.uint32(72).bool(message.isCollateralRatioValid);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.collRatio = reader.int64();
                    break;
                case 2:
                    message.feeRatio = reader.int64();
                    break;
                case 3:
                    message.efFeeRatio = reader.int64();
                    break;
                case 4:
                    message.bonusRateRecoll = reader.int64();
                    break;
                case 5:
                    message.distrEpochIdentifier = reader.string();
                    break;
                case 6:
                    message.adjustmentStep = reader.int64();
                    break;
                case 7:
                    message.priceLowerBound = reader.int64();
                    break;
                case 8:
                    message.priceUpperBound = reader.int64();
                    break;
                case 9:
                    message.isCollateralRatioValid = reader.bool();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            collRatio: isSet(object.collRatio) ? long_1.default.fromValue(object.collRatio) : long_1.default.ZERO,
            feeRatio: isSet(object.feeRatio) ? long_1.default.fromValue(object.feeRatio) : long_1.default.ZERO,
            efFeeRatio: isSet(object.efFeeRatio)
                ? long_1.default.fromValue(object.efFeeRatio)
                : long_1.default.ZERO,
            bonusRateRecoll: isSet(object.bonusRateRecoll)
                ? long_1.default.fromValue(object.bonusRateRecoll)
                : long_1.default.ZERO,
            distrEpochIdentifier: isSet(object.distrEpochIdentifier)
                ? String(object.distrEpochIdentifier)
                : "",
            adjustmentStep: isSet(object.adjustmentStep)
                ? long_1.default.fromValue(object.adjustmentStep)
                : long_1.default.ZERO,
            priceLowerBound: isSet(object.priceLowerBound)
                ? long_1.default.fromValue(object.priceLowerBound)
                : long_1.default.ZERO,
            priceUpperBound: isSet(object.priceUpperBound)
                ? long_1.default.fromValue(object.priceUpperBound)
                : long_1.default.ZERO,
            isCollateralRatioValid: isSet(object.isCollateralRatioValid)
                ? Boolean(object.isCollateralRatioValid)
                : false,
        };
    },
    toJSON(message) {
        const obj = {};
        message.collRatio !== undefined &&
            (obj.collRatio = (message.collRatio || long_1.default.ZERO).toString());
        message.feeRatio !== undefined &&
            (obj.feeRatio = (message.feeRatio || long_1.default.ZERO).toString());
        message.efFeeRatio !== undefined &&
            (obj.efFeeRatio = (message.efFeeRatio || long_1.default.ZERO).toString());
        message.bonusRateRecoll !== undefined &&
            (obj.bonusRateRecoll = (message.bonusRateRecoll || long_1.default.ZERO).toString());
        message.distrEpochIdentifier !== undefined &&
            (obj.distrEpochIdentifier = message.distrEpochIdentifier);
        message.adjustmentStep !== undefined &&
            (obj.adjustmentStep = (message.adjustmentStep || long_1.default.ZERO).toString());
        message.priceLowerBound !== undefined &&
            (obj.priceLowerBound = (message.priceLowerBound || long_1.default.ZERO).toString());
        message.priceUpperBound !== undefined &&
            (obj.priceUpperBound = (message.priceUpperBound || long_1.default.ZERO).toString());
        message.isCollateralRatioValid !== undefined &&
            (obj.isCollateralRatioValid = message.isCollateralRatioValid);
        return obj;
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseParams();
        message.collRatio =
            object.collRatio !== undefined && object.collRatio !== null
                ? long_1.default.fromValue(object.collRatio)
                : long_1.default.ZERO;
        message.feeRatio =
            object.feeRatio !== undefined && object.feeRatio !== null
                ? long_1.default.fromValue(object.feeRatio)
                : long_1.default.ZERO;
        message.efFeeRatio =
            object.efFeeRatio !== undefined && object.efFeeRatio !== null
                ? long_1.default.fromValue(object.efFeeRatio)
                : long_1.default.ZERO;
        message.bonusRateRecoll =
            object.bonusRateRecoll !== undefined && object.bonusRateRecoll !== null
                ? long_1.default.fromValue(object.bonusRateRecoll)
                : long_1.default.ZERO;
        message.distrEpochIdentifier = (_a = object.distrEpochIdentifier) !== null && _a !== void 0 ? _a : "";
        message.adjustmentStep =
            object.adjustmentStep !== undefined && object.adjustmentStep !== null
                ? long_1.default.fromValue(object.adjustmentStep)
                : long_1.default.ZERO;
        message.priceLowerBound =
            object.priceLowerBound !== undefined && object.priceLowerBound !== null
                ? long_1.default.fromValue(object.priceLowerBound)
                : long_1.default.ZERO;
        message.priceUpperBound =
            object.priceUpperBound !== undefined && object.priceUpperBound !== null
                ? long_1.default.fromValue(object.priceUpperBound)
                : long_1.default.ZERO;
        message.isCollateralRatioValid = (_b = object.isCollateralRatioValid) !== null && _b !== void 0 ? _b : false;
        return message;
    },
};
if (_m0.util.Long !== long_1.default) {
    _m0.util.Long = long_1.default;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
