// @nibiruchain/protojs/dist/nibiru/spot/v1/params.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Params = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const coin_1 = require("../../../cosmos/base/v1beta1/coin");
exports.protobufPackage = "nibiru.spot.v1";
function createBaseParams() {
    return {
        startingPoolNumber: long_1.default.UZERO,
        poolCreationFee: [],
        whitelistedAsset: [],
    };
}
exports.Params = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (!message.startingPoolNumber.isZero()) {
            writer.uint32(8).uint64(message.startingPoolNumber);
        }
        for (const v of message.poolCreationFee) {
            coin_1.Coin.encode(v, writer.uint32(18).fork()).ldelim();
        }
        for (const v of message.whitelistedAsset) {
            writer.uint32(26).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.startingPoolNumber = reader.uint64();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.poolCreationFee.push(coin_1.Coin.decode(reader, reader.uint32()));
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.whitelistedAsset.push(reader.string());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            startingPoolNumber: isSet(object.startingPoolNumber)
                ? long_1.default.fromValue(object.startingPoolNumber)
                : long_1.default.UZERO,
            poolCreationFee: Array.isArray(object === null || object === void 0 ? void 0 : object.poolCreationFee)
                ? object.poolCreationFee.map((e) => coin_1.Coin.fromJSON(e))
                : [],
            whitelistedAsset: Array.isArray(object === null || object === void 0 ? void 0 : object.whitelistedAsset)
                ? object.whitelistedAsset.map((e) => String(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        message.startingPoolNumber !== undefined &&
            (obj.startingPoolNumber = (message.startingPoolNumber || long_1.default.UZERO).toString());
        if (message.poolCreationFee) {
            obj.poolCreationFee = message.poolCreationFee.map((e) => e ? coin_1.Coin.toJSON(e) : undefined);
        }
        else {
            obj.poolCreationFee = [];
        }
        if (message.whitelistedAsset) {
            obj.whitelistedAsset = message.whitelistedAsset.map((e) => e);
        }
        else {
            obj.whitelistedAsset = [];
        }
        return obj;
    },
    create(base) {
        return exports.Params.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseParams();
        message.startingPoolNumber =
            object.startingPoolNumber !== undefined &&
                object.startingPoolNumber !== null
                ? long_1.default.fromValue(object.startingPoolNumber)
                : long_1.default.UZERO;
        message.poolCreationFee =
            ((_a = object.poolCreationFee) === null || _a === void 0 ? void 0 : _a.map((e) => coin_1.Coin.fromPartial(e))) || [];
        message.whitelistedAsset = ((_b = object.whitelistedAsset) === null || _b === void 0 ? void 0 : _b.map((e) => e)) || [];
        return message;
    },
};
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
