// @nibiruchain/nibijs/dist/msg/perp.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isMsgDonateToEcosystemFundEncodeObject = exports.isMsgClosePositionEncodeObject = exports.isMsgOpenPositionEncodeObject = exports.isMsgMultiLiquidateEncodeObject = exports.isMsgRemoveMarginEncodeObject = exports.isMsgAddMarginEncodeObject = exports.perpTypes = exports.PERP_MSG_TYPE_URLS = void 0;
const tx_1 = require("@nibiruchain/protojs/dist/perp/v1/tx");
exports.PERP_MSG_TYPE_URLS = {
    MsgAddMargin: `/${tx_1.protobufPackage}.MsgAddMargin`,
    MsgRemoveMargin: `/${tx_1.protobufPackage}.MsgRemoveMargin`,
    MsgMultiLiquidate: `/${tx_1.protobufPackage}.MsgMultiLiquidate`,
    MsgOpenPosition: `/${tx_1.protobufPackage}.MsgOpenPosition`,
    MsgClosePosition: `/${tx_1.protobufPackage}.MsgClosePosition`,
    MsgDonateToEcosystemFund: `/${tx_1.protobufPackage}.MsgDonateToEcosystemFund`,
};
exports.perpTypes = [
    [exports.PERP_MSG_TYPE_URLS.MsgAddMargin, tx_1.MsgAddMargin],
    [exports.PERP_MSG_TYPE_URLS.MsgRemoveMargin, tx_1.MsgRemoveMargin],
    [exports.PERP_MSG_TYPE_URLS.MsgMultiLiquidate, tx_1.MsgMultiLiquidate],
    [exports.PERP_MSG_TYPE_URLS.MsgOpenPosition, tx_1.MsgOpenPosition],
    [exports.PERP_MSG_TYPE_URLS.MsgClosePosition, tx_1.MsgClosePosition],
    [exports.PERP_MSG_TYPE_URLS.MsgDonateToEcosystemFund, tx_1.MsgDonateToEcosystemFund],
];
function isMsgAddMarginEncodeObject(encodeObject) {
    return (encodeObject.typeUrl ===
        exports.PERP_MSG_TYPE_URLS.MsgAddMargin);
}
exports.isMsgAddMarginEncodeObject = isMsgAddMarginEncodeObject;
function isMsgRemoveMarginEncodeObject(encodeObject) {
    return (encodeObject.typeUrl ===
        exports.PERP_MSG_TYPE_URLS.MsgRemoveMargin);
}
exports.isMsgRemoveMarginEncodeObject = isMsgRemoveMarginEncodeObject;
function isMsgMultiLiquidateEncodeObject(encodeObject) {
    return (encodeObject.typeUrl ===
        exports.PERP_MSG_TYPE_URLS.MsgMultiLiquidate);
}
exports.isMsgMultiLiquidateEncodeObject = isMsgMultiLiquidateEncodeObject;
function isMsgOpenPositionEncodeObject(encodeObject) {
    return (encodeObject.typeUrl ===
        exports.PERP_MSG_TYPE_URLS.MsgOpenPosition);
}
exports.isMsgOpenPositionEncodeObject = isMsgOpenPositionEncodeObject;
function isMsgClosePositionEncodeObject(encodeObject) {
    return (encodeObject.typeUrl ===
        exports.PERP_MSG_TYPE_URLS.MsgClosePosition);
}
exports.isMsgClosePositionEncodeObject = isMsgClosePositionEncodeObject;
function isMsgDonateToEcosystemFundEncodeObject(encodeObject) {
    return (encodeObject.typeUrl ===
        exports.PERP_MSG_TYPE_URLS.MsgDonateToEcosystemFund);
}
exports.isMsgDonateToEcosystemFundEncodeObject = isMsgDonateToEcosystemFundEncodeObject;
