// @nibiruchain/nibijs/dist/msg/spot.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isMsgSwapAssetsEncodeObject = exports.isMsgExitPoolEncodeObject = exports.isMsgJoinPoolEncodeObject = exports.isMsgCreatePoolEncodeObject = exports.spotTypes = exports.SPOT_MSG_TYPE_URLS = void 0;
const tx_1 = require("@nibiruchain/protojs/dist/spot/v1/tx");
exports.SPOT_MSG_TYPE_URLS = {
    MsgCreatePool: `/${tx_1.protobufPackage}.MsgCreatePool`,
    MsgJoinPool: `/${tx_1.protobufPackage}.MsgJoinPool`,
    MsgExitPool: `/${tx_1.protobufPackage}.MsgExitPool`,
    MsgSwapAssets: `/${tx_1.protobufPackage}.MsgSwapAssets`,
};
exports.spotTypes = [
    [exports.SPOT_MSG_TYPE_URLS.MsgCreatePool, tx_1.MsgCreatePool],
    [exports.SPOT_MSG_TYPE_URLS.MsgJoinPool, tx_1.MsgJoinPool],
    [exports.SPOT_MSG_TYPE_URLS.MsgExitPool, tx_1.MsgExitPool],
    [exports.SPOT_MSG_TYPE_URLS.MsgSwapAssets, tx_1.MsgSwapAssets],
];
function isMsgCreatePoolEncodeObject(encodeObject) {
    return encodeObject.typeUrl === exports.SPOT_MSG_TYPE_URLS.MsgCreatePool;
}
exports.isMsgCreatePoolEncodeObject = isMsgCreatePoolEncodeObject;
function isMsgJoinPoolEncodeObject(encodeObject) {
    return encodeObject.typeUrl === exports.SPOT_MSG_TYPE_URLS.MsgJoinPool;
}
exports.isMsgJoinPoolEncodeObject = isMsgJoinPoolEncodeObject;
function isMsgExitPoolEncodeObject(encodeObject) {
    return encodeObject.typeUrl === exports.SPOT_MSG_TYPE_URLS.MsgExitPool;
}
exports.isMsgExitPoolEncodeObject = isMsgExitPoolEncodeObject;
function isMsgSwapAssetsEncodeObject(encodeObject) {
    return encodeObject.typeUrl === exports.SPOT_MSG_TYPE_URLS.MsgSwapAssets;
}
exports.isMsgSwapAssetsEncodeObject = isMsgSwapAssetsEncodeObject;
