// @nibiruchain/protojs/dist/nibiru/epochs/v1/state.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EpochInfo = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const duration_1 = require("../../../google/protobuf/duration");
const timestamp_1 = require("../../../google/protobuf/timestamp");
exports.protobufPackage = "nibiru.epochs.v1";
function createBaseEpochInfo() {
    return {
        identifier: "",
        startTime: undefined,
        duration: undefined,
        currentEpoch: long_1.default.UZERO,
        currentEpochStartTime: undefined,
        epochCountingStarted: false,
        currentEpochStartHeight: long_1.default.ZERO,
    };
}
exports.EpochInfo = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.identifier !== "") {
            writer.uint32(10).string(message.identifier);
        }
        if (message.startTime !== undefined) {
            timestamp_1.Timestamp.encode(toTimestamp(message.startTime), writer.uint32(18).fork()).ldelim();
        }
        if (message.duration !== undefined) {
            duration_1.Duration.encode(message.duration, writer.uint32(26).fork()).ldelim();
        }
        if (!message.currentEpoch.isZero()) {
            writer.uint32(32).uint64(message.currentEpoch);
        }
        if (message.currentEpochStartTime !== undefined) {
            timestamp_1.Timestamp.encode(toTimestamp(message.currentEpochStartTime), writer.uint32(42).fork()).ldelim();
        }
        if (message.epochCountingStarted === true) {
            writer.uint32(48).bool(message.epochCountingStarted);
        }
        if (!message.currentEpochStartHeight.isZero()) {
            writer.uint32(56).int64(message.currentEpochStartHeight);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEpochInfo();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.identifier = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.startTime = fromTimestamp(timestamp_1.Timestamp.decode(reader, reader.uint32()));
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.duration = duration_1.Duration.decode(reader, reader.uint32());
                    continue;
                case 4:
                    if (tag !== 32) {
                        break;
                    }
                    message.currentEpoch = reader.uint64();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.currentEpochStartTime = fromTimestamp(timestamp_1.Timestamp.decode(reader, reader.uint32()));
                    continue;
                case 6:
                    if (tag !== 48) {
                        break;
                    }
                    message.epochCountingStarted = reader.bool();
                    continue;
                case 7:
                    if (tag !== 56) {
                        break;
                    }
                    message.currentEpochStartHeight = reader.int64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            identifier: isSet(object.identifier) ? String(object.identifier) : "",
            startTime: isSet(object.startTime)
                ? fromJsonTimestamp(object.startTime)
                : undefined,
            duration: isSet(object.duration)
                ? duration_1.Duration.fromJSON(object.duration)
                : undefined,
            currentEpoch: isSet(object.currentEpoch)
                ? long_1.default.fromValue(object.currentEpoch)
                : long_1.default.UZERO,
            currentEpochStartTime: isSet(object.currentEpochStartTime)
                ? fromJsonTimestamp(object.currentEpochStartTime)
                : undefined,
            epochCountingStarted: isSet(object.epochCountingStarted)
                ? Boolean(object.epochCountingStarted)
                : false,
            currentEpochStartHeight: isSet(object.currentEpochStartHeight)
                ? long_1.default.fromValue(object.currentEpochStartHeight)
                : long_1.default.ZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        message.identifier !== undefined && (obj.identifier = message.identifier);
        message.startTime !== undefined &&
            (obj.startTime = message.startTime.toISOString());
        message.duration !== undefined &&
            (obj.duration = message.duration
                ? duration_1.Duration.toJSON(message.duration)
                : undefined);
        message.currentEpoch !== undefined &&
            (obj.currentEpoch = (message.currentEpoch || long_1.default.UZERO).toString());
        message.currentEpochStartTime !== undefined &&
            (obj.currentEpochStartTime = message.currentEpochStartTime.toISOString());
        message.epochCountingStarted !== undefined &&
            (obj.epochCountingStarted = message.epochCountingStarted);
        message.currentEpochStartHeight !== undefined &&
            (obj.currentEpochStartHeight = (message.currentEpochStartHeight || long_1.default.ZERO).toString());
        return obj;
    },
    create(base) {
        return exports.EpochInfo.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c, _d;
        const message = createBaseEpochInfo();
        message.identifier = (_a = object.identifier) !== null && _a !== void 0 ? _a : "";
        message.startTime = (_b = object.startTime) !== null && _b !== void 0 ? _b : undefined;
        message.duration =
            object.duration !== undefined && object.duration !== null
                ? duration_1.Duration.fromPartial(object.duration)
                : undefined;
        message.currentEpoch =
            object.currentEpoch !== undefined && object.currentEpoch !== null
                ? long_1.default.fromValue(object.currentEpoch)
                : long_1.default.UZERO;
        message.currentEpochStartTime = (_c = object.currentEpochStartTime) !== null && _c !== void 0 ? _c : undefined;
        message.epochCountingStarted = (_d = object.epochCountingStarted) !== null && _d !== void 0 ? _d : false;
        message.currentEpochStartHeight =
            object.currentEpochStartHeight !== undefined &&
                object.currentEpochStartHeight !== null
                ? long_1.default.fromValue(object.currentEpochStartHeight)
                : long_1.default.ZERO;
        return message;
    },
};
function toTimestamp(date) {
    const seconds = numberToLong(date.getTime() / 1000);
    const nanos = (date.getTime() % 1000) * 1000000;
    return { seconds, nanos };
}
function fromTimestamp(t) {
    let millis = (t.seconds.toNumber() || 0) * 1000;
    millis += (t.nanos || 0) / 1000000;
    return new Date(millis);
}
function fromJsonTimestamp(o) {
    if (o instanceof Date) {
        return o;
    }
    else if (typeof o === "string") {
        return new Date(o);
    }
    else {
        return fromTimestamp(timestamp_1.Timestamp.fromJSON(o));
    }
}
function numberToLong(number) {
    return long_1.default.fromNumber(number);
}
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
