// @nibiruchain/protojs/dist/nibiru/inflation/v1/genesis.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Params = exports.GenesisState = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const inflation_1 = require("./inflation");
exports.protobufPackage = "nibiru.inflation.v1";
function createBaseGenesisState() {
    return { params: undefined, period: long_1.default.UZERO, skippedEpochs: long_1.default.UZERO };
}
exports.GenesisState = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.params !== undefined) {
            exports.Params.encode(message.params, writer.uint32(10).fork()).ldelim();
        }
        if (!message.period.isZero()) {
            writer.uint32(16).uint64(message.period);
        }
        if (!message.skippedEpochs.isZero()) {
            writer.uint32(24).uint64(message.skippedEpochs);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGenesisState();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.params = exports.Params.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.period = reader.uint64();
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.skippedEpochs = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            params: isSet(object.params) ? exports.Params.fromJSON(object.params) : undefined,
            period: isSet(object.period) ? long_1.default.fromValue(object.period) : long_1.default.UZERO,
            skippedEpochs: isSet(object.skippedEpochs)
                ? long_1.default.fromValue(object.skippedEpochs)
                : long_1.default.UZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        message.params !== undefined &&
            (obj.params = message.params ? exports.Params.toJSON(message.params) : undefined);
        message.period !== undefined &&
            (obj.period = (message.period || long_1.default.UZERO).toString());
        message.skippedEpochs !== undefined &&
            (obj.skippedEpochs = (message.skippedEpochs || long_1.default.UZERO).toString());
        return obj;
    },
    create(base) {
        return exports.GenesisState.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        const message = createBaseGenesisState();
        message.params =
            object.params !== undefined && object.params !== null
                ? exports.Params.fromPartial(object.params)
                : undefined;
        message.period =
            object.period !== undefined && object.period !== null
                ? long_1.default.fromValue(object.period)
                : long_1.default.UZERO;
        message.skippedEpochs =
            object.skippedEpochs !== undefined && object.skippedEpochs !== null
                ? long_1.default.fromValue(object.skippedEpochs)
                : long_1.default.UZERO;
        return message;
    },
};
function createBaseParams() {
    return {
        inflationEnabled: false,
        exponentialCalculation: undefined,
        inflationDistribution: undefined,
        epochsPerPeriod: long_1.default.UZERO,
    };
}
exports.Params = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.inflationEnabled === true) {
            writer.uint32(8).bool(message.inflationEnabled);
        }
        if (message.exponentialCalculation !== undefined) {
            inflation_1.ExponentialCalculation.encode(message.exponentialCalculation, writer.uint32(18).fork()).ldelim();
        }
        if (message.inflationDistribution !== undefined) {
            inflation_1.InflationDistribution.encode(message.inflationDistribution, writer.uint32(26).fork()).ldelim();
        }
        if (!message.epochsPerPeriod.isZero()) {
            writer.uint32(32).uint64(message.epochsPerPeriod);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.inflationEnabled = reader.bool();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.exponentialCalculation = inflation_1.ExponentialCalculation.decode(reader, reader.uint32());
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.inflationDistribution = inflation_1.InflationDistribution.decode(reader, reader.uint32());
                    continue;
                case 4:
                    if (tag !== 32) {
                        break;
                    }
                    message.epochsPerPeriod = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            inflationEnabled: isSet(object.inflationEnabled)
                ? Boolean(object.inflationEnabled)
                : false,
            exponentialCalculation: isSet(object.exponentialCalculation)
                ? inflation_1.ExponentialCalculation.fromJSON(object.exponentialCalculation)
                : undefined,
            inflationDistribution: isSet(object.inflationDistribution)
                ? inflation_1.InflationDistribution.fromJSON(object.inflationDistribution)
                : undefined,
            epochsPerPeriod: isSet(object.epochsPerPeriod)
                ? long_1.default.fromValue(object.epochsPerPeriod)
                : long_1.default.UZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        message.inflationEnabled !== undefined &&
            (obj.inflationEnabled = message.inflationEnabled);
        message.exponentialCalculation !== undefined &&
            (obj.exponentialCalculation = message.exponentialCalculation
                ? inflation_1.ExponentialCalculation.toJSON(message.exponentialCalculation)
                : undefined);
        message.inflationDistribution !== undefined &&
            (obj.inflationDistribution = message.inflationDistribution
                ? inflation_1.InflationDistribution.toJSON(message.inflationDistribution)
                : undefined);
        message.epochsPerPeriod !== undefined &&
            (obj.epochsPerPeriod = (message.epochsPerPeriod || long_1.default.UZERO).toString());
        return obj;
    },
    create(base) {
        return exports.Params.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseParams();
        message.inflationEnabled = (_a = object.inflationEnabled) !== null && _a !== void 0 ? _a : false;
        message.exponentialCalculation =
            object.exponentialCalculation !== undefined &&
                object.exponentialCalculation !== null
                ? inflation_1.ExponentialCalculation.fromPartial(object.exponentialCalculation)
                : undefined;
        message.inflationDistribution =
            object.inflationDistribution !== undefined &&
                object.inflationDistribution !== null
                ? inflation_1.InflationDistribution.fromPartial(object.inflationDistribution)
                : undefined;
        message.epochsPerPeriod =
            object.epochsPerPeriod !== undefined && object.epochsPerPeriod !== null
                ? long_1.default.fromValue(object.epochsPerPeriod)
                : long_1.default.UZERO;
        return message;
    },
};
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
