// @nibiruchain/protojs/dist/nibiru/sudo/v1/event.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventUpdateSudoers = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const state_1 = require("./state");
exports.protobufPackage = "nibiru.sudo.v1";
function createBaseEventUpdateSudoers() {
    return { sudoers: undefined, action: "" };
}
exports.EventUpdateSudoers = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sudoers !== undefined) {
            state_1.Sudoers.encode(message.sudoers, writer.uint32(10).fork()).ldelim();
        }
        if (message.action !== "") {
            writer.uint32(18).string(message.action);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEventUpdateSudoers();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.sudoers = state_1.Sudoers.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.action = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sudoers: isSet(object.sudoers)
                ? state_1.Sudoers.fromJSON(object.sudoers)
                : undefined,
            action: isSet(object.action) ? String(object.action) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        message.sudoers !== undefined &&
            (obj.sudoers = message.sudoers
                ? state_1.Sudoers.toJSON(message.sudoers)
                : undefined);
        message.action !== undefined && (obj.action = message.action);
        return obj;
    },
    create(base) {
        return exports.EventUpdateSudoers.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseEventUpdateSudoers();
        message.sudoers =
            object.sudoers !== undefined && object.sudoers !== null
                ? state_1.Sudoers.fromPartial(object.sudoers)
                : undefined;
        message.action = (_a = object.action) !== null && _a !== void 0 ? _a : "";
        return message;
    },
};
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
