// @nibiruchain/protojs/dist/nibiru/sudo/v1/query.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryClientImpl = exports.QueryServiceName = exports.QuerySudoersResponse = exports.QuerySudoersRequest = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const state_1 = require("./state");
exports.protobufPackage = "nibiru.sudo.v1";
function createBaseQuerySudoersRequest() {
    return {};
}
exports.QuerySudoersRequest = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQuerySudoersRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.QuerySudoersRequest.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(_) {
        const message = createBaseQuerySudoersRequest();
        return message;
    },
};
function createBaseQuerySudoersResponse() {
    return { sudoers: undefined };
}
exports.QuerySudoersResponse = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sudoers !== undefined) {
            state_1.Sudoers.encode(message.sudoers, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQuerySudoersResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.sudoers = state_1.Sudoers.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sudoers: isSet(object.sudoers)
                ? state_1.Sudoers.fromJSON(object.sudoers)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        message.sudoers !== undefined &&
            (obj.sudoers = message.sudoers
                ? state_1.Sudoers.toJSON(message.sudoers)
                : undefined);
        return obj;
    },
    create(base) {
        return exports.QuerySudoersResponse.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        const message = createBaseQuerySudoersResponse();
        message.sudoers =
            object.sudoers !== undefined && object.sudoers !== null
                ? state_1.Sudoers.fromPartial(object.sudoers)
                : undefined;
        return message;
    },
};
exports.QueryServiceName = "nibiru.sudo.v1.Query";
class QueryClientImpl {
    constructor(rpc, opts) {
        this.service = (opts === null || opts === void 0 ? void 0 : opts.service) || exports.QueryServiceName;
        this.rpc = rpc;
        this.QuerySudoers = this.QuerySudoers.bind(this);
    }
    QuerySudoers(request) {
        const data = exports.QuerySudoersRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "QuerySudoers", data);
        return promise.then((data) => exports.QuerySudoersResponse.decode(minimal_1.default.Reader.create(data)));
    }
}
exports.QueryClientImpl = QueryClientImpl;
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
