// @nibiruchain/nibijs/dist/query/spot.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.setupSpotExtension = void 0;
const stargate_1 = require("@cosmjs/stargate");
const qpb = __importStar(require("@nibiruchain/protojs/dist/spot/v1/query"));
const chain_1 = require("../chain");
function transformPoolParams(pp) {
    if (pp) {
        pp.swapFee = (0, chain_1.fromSdkDec)(pp === null || pp === void 0 ? void 0 : pp.swapFee).toString();
        pp.exitFee = (0, chain_1.fromSdkDec)(pp === null || pp === void 0 ? void 0 : pp.exitFee).toString();
    }
    return pp;
}
function transformPool(p) {
    if (!p) {
        return p;
    }
    p.poolParams = transformPoolParams(p.poolParams);
    return p;
}
function setupSpotExtension(base) {
    const rpcClient = (0, stargate_1.createProtobufRpcClient)(base);
    const queryService = new qpb.QueryClientImpl(rpcClient);
    return {
        spot: {
            params: () => {
                const req = qpb.QueryParamsRequest.fromPartial({});
                return queryService.Params(req);
            },
            poolNumber: () => {
                const req = qpb.QueryPoolNumberRequest.fromPartial({});
                return queryService.PoolNumber(req);
            },
            pool: (poolId) => __awaiter(this, void 0, void 0, function* () {
                const req = qpb.QueryPoolRequest.fromPartial({ poolId });
                const resp = yield queryService.Pool(req);
                resp.pool = transformPool(resp.pool);
                return resp;
            }),
            pools: (pagination) => __awaiter(this, void 0, void 0, function* () {
                const req = qpb.QueryPoolsRequest.fromPartial({ pagination });
                const resp = yield queryService.Pools(req);
                resp.pools = resp.pools.map((p) => transformPool(p));
                return resp;
            }),
            poolParams: (poolId) => __awaiter(this, void 0, void 0, function* () {
                const req = qpb.QueryPoolParamsRequest.fromPartial({ poolId });
                const resp = yield queryService.PoolParams(req);
                resp.poolParams = transformPoolParams(resp.poolParams);
                return resp;
            }),
            numPools: () => {
                const req = qpb.QueryNumPoolsRequest.fromPartial({});
                return queryService.NumPools(req);
            },
            totalLiquidity: () => {
                const req = qpb.QueryTotalLiquidityRequest.fromPartial({});
                return queryService.TotalLiquidity(req);
            },
            totalPoolLiquidity: (poolId) => {
                const req = qpb.QueryTotalPoolLiquidityRequest.fromPartial({
                    poolId,
                });
                return queryService.TotalPoolLiquidity(req);
            },
            totalShares: (poolId) => {
                const req = qpb.QueryTotalSharesRequest.fromPartial({ poolId });
                return queryService.TotalShares(req);
            },
            spotPrice: (poolId, tokenInDenom, tokenOutDenom) => {
                const req = qpb.QuerySpotPriceRequest.fromPartial({
                    poolId,
                    tokenInDenom,
                    tokenOutDenom,
                });
                return queryService.SpotPrice(req);
            },
            estimateSwapExactAmountIn: (poolId, tokenOutDenom, tokenIn) => {
                const req = qpb.QuerySwapExactAmountInRequest.fromPartial({
                    poolId,
                    tokenOutDenom,
                    tokenIn,
                });
                return queryService.EstimateSwapExactAmountIn(req);
            },
            estimateSwapExactAmountOut: (poolId, tokenInDenom, tokenOut) => {
                const req = qpb.QuerySwapExactAmountOutRequest.fromPartial({
                    poolId,
                    tokenInDenom,
                    tokenOut,
                });
                return queryService.EstimateSwapExactAmountOut(req);
            },
            estimateJoinExactAmountIn: (poolId, tokensIn) => {
                const req = qpb.QueryJoinExactAmountInRequest.fromPartial({
                    poolId,
                    tokensIn,
                });
                return queryService.EstimateJoinExactAmountIn(req);
            },
            estimateJoinExactAmountOut: (poolId) => {
                const req = qpb.QueryJoinExactAmountOutRequest.fromPartial({
                    poolId,
                });
                return queryService.EstimateJoinExactAmountOut(req);
            },
            estimateExitExactAmountIn: (poolId, poolSharesIn) => {
                const req = qpb.QueryExitExactAmountInRequest.fromPartial({
                    poolId,
                    poolSharesIn: poolSharesIn.toString(),
                });
                return queryService.EstimateExitExactAmountIn(req);
            },
            estimateExitExactAmountOut: (poolId) => {
                const req = qpb.QueryExitExactAmountOutRequest.fromPartial({
                    poolId,
                });
                return queryService.EstimateExitExactAmountOut(req);
            },
        },
    };
}
exports.setupSpotExtension = setupSpotExtension;
