// @nibiruchain/protojs/dist/epochs/event.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventEpochEnd = exports.EventEpochStart = exports.protobufPackage = void 0;
/* eslint-disable */
const timestamp_1 = require("../google/protobuf/timestamp");
const long_1 = __importDefault(require("long"));
const _m0 = __importStar(require("protobufjs/minimal"));
exports.protobufPackage = "nibiru.epochs.v1";
function createBaseEventEpochStart() {
    return { epochNumber: long_1.default.UZERO, epochStartTime: undefined };
}
exports.EventEpochStart = {
    encode(message, writer = _m0.Writer.create()) {
        if (!message.epochNumber.isZero()) {
            writer.uint32(8).uint64(message.epochNumber);
        }
        if (message.epochStartTime !== undefined) {
            timestamp_1.Timestamp.encode(toTimestamp(message.epochStartTime), writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEventEpochStart();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.epochNumber = reader.uint64();
                    break;
                case 2:
                    message.epochStartTime = fromTimestamp(timestamp_1.Timestamp.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            epochNumber: isSet(object.epochNumber)
                ? long_1.default.fromValue(object.epochNumber)
                : long_1.default.UZERO,
            epochStartTime: isSet(object.epochStartTime)
                ? fromJsonTimestamp(object.epochStartTime)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        message.epochNumber !== undefined &&
            (obj.epochNumber = (message.epochNumber || long_1.default.UZERO).toString());
        message.epochStartTime !== undefined &&
            (obj.epochStartTime = message.epochStartTime.toISOString());
        return obj;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseEventEpochStart();
        message.epochNumber =
            object.epochNumber !== undefined && object.epochNumber !== null
                ? long_1.default.fromValue(object.epochNumber)
                : long_1.default.UZERO;
        message.epochStartTime = (_a = object.epochStartTime) !== null && _a !== void 0 ? _a : undefined;
        return message;
    },
};
function createBaseEventEpochEnd() {
    return { epochNumber: long_1.default.UZERO };
}
exports.EventEpochEnd = {
    encode(message, writer = _m0.Writer.create()) {
        if (!message.epochNumber.isZero()) {
            writer.uint32(8).uint64(message.epochNumber);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEventEpochEnd();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.epochNumber = reader.uint64();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            epochNumber: isSet(object.epochNumber)
                ? long_1.default.fromValue(object.epochNumber)
                : long_1.default.UZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        message.epochNumber !== undefined &&
            (obj.epochNumber = (message.epochNumber || long_1.default.UZERO).toString());
        return obj;
    },
    fromPartial(object) {
        const message = createBaseEventEpochEnd();
        message.epochNumber =
            object.epochNumber !== undefined && object.epochNumber !== null
                ? long_1.default.fromValue(object.epochNumber)
                : long_1.default.UZERO;
        return message;
    },
};
function toTimestamp(date) {
    const seconds = numberToLong(date.getTime() / 1000);
    const nanos = (date.getTime() % 1000) * 1000000;
    return { seconds, nanos };
}
function fromTimestamp(t) {
    let millis = t.seconds.toNumber() * 1000;
    millis += t.nanos / 1000000;
    return new Date(millis);
}
function fromJsonTimestamp(o) {
    if (o instanceof Date) {
        return o;
    }
    else if (typeof o === "string") {
        return new Date(o);
    }
    else {
        return fromTimestamp(timestamp_1.Timestamp.fromJSON(o));
    }
}
function numberToLong(number) {
    return long_1.default.fromNumber(number);
}
if (_m0.util.Long !== long_1.default) {
    _m0.util.Long = long_1.default;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
