// @nibiruchain/protojs/dist/nibiru/perp/v2/genesis.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenesisState = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const state_1 = require("./state");
exports.protobufPackage = "nibiru.perp.v2";
function createBaseGenesisState() {
    return { markets: [], amms: [], positions: [], reserveSnapshots: [] };
}
exports.GenesisState = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.markets) {
            state_1.Market.encode(v, writer.uint32(18).fork()).ldelim();
        }
        for (const v of message.amms) {
            state_1.AMM.encode(v, writer.uint32(26).fork()).ldelim();
        }
        for (const v of message.positions) {
            state_1.Position.encode(v, writer.uint32(34).fork()).ldelim();
        }
        for (const v of message.reserveSnapshots) {
            state_1.ReserveSnapshot.encode(v, writer.uint32(42).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGenesisState();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.markets.push(state_1.Market.decode(reader, reader.uint32()));
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.amms.push(state_1.AMM.decode(reader, reader.uint32()));
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.positions.push(state_1.Position.decode(reader, reader.uint32()));
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.reserveSnapshots.push(state_1.ReserveSnapshot.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            markets: Array.isArray(object === null || object === void 0 ? void 0 : object.markets)
                ? object.markets.map((e) => state_1.Market.fromJSON(e))
                : [],
            amms: Array.isArray(object === null || object === void 0 ? void 0 : object.amms)
                ? object.amms.map((e) => state_1.AMM.fromJSON(e))
                : [],
            positions: Array.isArray(object === null || object === void 0 ? void 0 : object.positions)
                ? object.positions.map((e) => state_1.Position.fromJSON(e))
                : [],
            reserveSnapshots: Array.isArray(object === null || object === void 0 ? void 0 : object.reserveSnapshots)
                ? object.reserveSnapshots.map((e) => state_1.ReserveSnapshot.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.markets) {
            obj.markets = message.markets.map((e) => e ? state_1.Market.toJSON(e) : undefined);
        }
        else {
            obj.markets = [];
        }
        if (message.amms) {
            obj.amms = message.amms.map((e) => (e ? state_1.AMM.toJSON(e) : undefined));
        }
        else {
            obj.amms = [];
        }
        if (message.positions) {
            obj.positions = message.positions.map((e) => e ? state_1.Position.toJSON(e) : undefined);
        }
        else {
            obj.positions = [];
        }
        if (message.reserveSnapshots) {
            obj.reserveSnapshots = message.reserveSnapshots.map((e) => e ? state_1.ReserveSnapshot.toJSON(e) : undefined);
        }
        else {
            obj.reserveSnapshots = [];
        }
        return obj;
    },
    create(base) {
        return exports.GenesisState.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c, _d;
        const message = createBaseGenesisState();
        message.markets = ((_a = object.markets) === null || _a === void 0 ? void 0 : _a.map((e) => state_1.Market.fromPartial(e))) || [];
        message.amms = ((_b = object.amms) === null || _b === void 0 ? void 0 : _b.map((e) => state_1.AMM.fromPartial(e))) || [];
        message.positions =
            ((_c = object.positions) === null || _c === void 0 ? void 0 : _c.map((e) => state_1.Position.fromPartial(e))) || [];
        message.reserveSnapshots =
            ((_d = object.reserveSnapshots) === null || _d === void 0 ? void 0 : _d.map((e) => state_1.ReserveSnapshot.fromPartial(e))) || [];
        return message;
    },
};
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
